submodules/wizard/wizard.js

```diff
@@ -1370,6 +1369,396 @@ define(function(require) {
+		 */
+		wizardSubmitGetFormattedServicePlan: function(wizardData) {
+			var self = this,
+				selectedPlanIds = wizardData.servicePlan.selectedPlanIds;
+
+			if (_.isEmpty(selectedPlanIds)) {
+				return null;
+			}
+
+			return {
+				add: selectedPlanIds
+			};
+		},
+
+		/**
+		 * Build an object that contain the account limits to submit to the API, from the wizard data
+		 * @param  {Object} wizardData  Wizard's data
+		 * @returns  {Object}  Plan limits
+		 */
+		wizardSubmitGetFormattedLimits: function(wizardData) {
+			var self = this,
+				limits = {
+					allow_prepay: true
+				};
+
+			_.each(wizardData.usageAndCallRestrictions.trunkLimits, function(value, trunkType) {
+				limits[trunkType + '_trunks'] = value;
+			});
+
+			return limits;
+		},
+
+		wizardSubmitNotifyErrors: function(error) {
+			var self = this,
+				errorCollection = error.error,
+				isAnyUserError = _.some(errorCollection, function(error, key) {
+					return _.startsWith(key, 'user');
+				}),
+				limitsErrorCode = _.get(errorCollection, 'limits.error'),
+				planErrorCode = _.get(errorCollection, 'plan.error'),
+				toastOptions = {
+					timeOut: self.appFlags.wizard.toastTimeout
+				};
+
+			if (_.get(error, 'type') === 'account') {
+				// Nor the account nor any of its related parts were created
+				monster.ui.toast({
+					type: 'error',
+					message: self.i18n.active().toastrMessages.newAccount.accountError,
+					options: toastOptions
+				});
+
+				return;
+			}
+
+			// If the account creation did not fail, there were errors in any of the features
+
+			// User errors
+			if (isAnyUserError) {
+				monster.ui.toast({
+					type: 'warning',
+					message: self.i18n.active().toastrMessages.newAccount.adminError,
+					options: toastOptions
+				});
+			}
+
+			// Limits errors
+			if (limitsErrorCode) {
+				if (limitsErrorCode === '403') {
+					monster.ui.toast({
+						type: 'warning',
+						message: self.i18n.active().toastrMessages.newAccount.forbiddenLimitsError,
+						options: toastOptions
+					});
+				} else if (limitsErrorCode !== '402') { // Only show error if error isn't a 402, because a 402 is handled generically
+					monster.ui.toast({
+						type: 'warning',
+						message: self.i18n.active().toastrMessages.newAccount.limitsError,
+						options: toastOptions
+					});
+				}
+			}
+
+			// Plan errors
+			if (planErrorCode) {
+				if (planErrorCode === '403') {
+					monster.ui.toast({
+						type: 'warning',
+						message: self.i18n.active().toastrMessages.newAccount.forbiddenPlanError,
+						options: toastOptions
+					});
+				} else if (planErrorCode !== '402') { // Only show error if error isn't a 402, because a 402 is handled generically
+					monster.ui.toast({
+						type: 'warning',
+						message: self.i18n.active().toastrMessages.newAccount.planError,
+						options: toastOptions
+					});
+				}
+			}
+
+			// Credit errors
+			if (_.has(errorCollection, 'credit')) {
+				monster.ui.toast({
```

Comment:
Looks like a lot of repetitive invokation of `monster.ui#toast`. Instead, you could save the message in a toastData variable and conditionally invoke the toast:

```js
var toastData = {};

if (isConditionTrue) {
    toastData = {
        message: 'my error message'
    };
}

// ...

if (!_isEmpty(toastData)) {
    monster.ui.toast(_.merge({
        type: 'warning',
        options: toastOptions
    }, toastData));
}
```

Reply:
I kinda copied that logic from the old account wizard, along with the error messages. It displays a toast per each error. Should I display all the messages in a single toast instead?

![image](https://user-images.githubusercontent.com/893877/62310161-22703480-b446-11e9-8ed9-ebec4e00efd1.png)

Also, I currently display most of them as warnings, because the account was successfully created at that point, although they are actual errors from the API requests. Is it OK? Originally most of them were displayed as `info`, and just one or two of them were displayed as `error`.

Reply:
Ah, in that case you can use an array to collect the different errors and invoke `monster.ui#toast` for each item.

And warnings make sense to indicate that something went wrong yet the account was still created.